pkg/runctl/controller.go

```diff
@@ -90,9 +96,29 @@ func NewController(factory k8s.ClientFactory, metrics metrics.Metrics) *Controll
 			controller.handleTektonTaskRun(new)
 		},
 	})
+
 	return controller
 }
 
+//meterCurrentPipelineStatus writes the current Duration of Pipeline States in a histogram
+func (c *Controller) meterCurrentPipelineStatus() {
+	klog.V(4).Infof("Current metrics Observation still alive")
+	objs := c.pipelineRunStore.List()
+	for _, obj := range objs {
+		pipelineRun, ok := obj.(*api.PipelineRun)
+		if !ok {
+			klog.V(4).Infof("failed to cast cached object of type %T to a PipelineRun", obj)
```

Comment:
I see. `Verbose.Error()` or `Verbose.ErrorS()` would be correct then. For those an `error` object must be provided:

```go
klog.V(4).Errorf(
	fmt.Errorf("object of type %T cannot be cast to %T", obj, (*api.PipelineRun)(nil)),
	"Found unexpected object in pipeline run cache",
)
```

In this case it would be even better to crash, as there must be programming error behind, and a level 4 message nobody will recognize:

```go
pipelineRun := obj.(*api.PipelineRun)
```

Reply:
Verbose.Error() or Verbose.ErrorS() is only available with k8s.io/klog/v2 v2.3.0. I suggest to update it once in PR #147  and we decided against it.
Anyway I can let it crash here, but for the second message with the failing observation I would have to use the klog.Error(), the Infof to use a log level or update the lib.

Reply:
Workaround as long as we can't upgrade to klog v2.3+:

```go
if klog.V(...).enabled() {
    klog.Errorf(...)
}
```

Reply:
ok, done